go/chat/unfurl/unfurler.go

```diff
@@ -203,13 +209,27 @@ func (u *Unfurler) makeBaseUnfurlMessage(ctx context.Context, fromMsg chat1.Mess
 func (u *Unfurler) UnfurlAndSend(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID,
 	msg chat1.MessageUnboxed) {
 	defer u.Trace(ctx, func() error { return nil }, "UnfurlAndSend")()
+	// early out for errors
+	if !msg.IsValid() {
+		return
+	}
 	// get URL hits
 	hits := u.extractURLs(ctx, uid, msg)
 	if len(hits) == 0 {
 		return
 	}
+	// get a map for all the URLs we have already unfurled
+	prevUnfurled := make(map[string]bool)
+	for _, u := range msg.Valid().Unfurls {
```

Comment:
isn't there a race for this to show up? maybe also check the urls present in `unfurlMap` in case a dup is pending?